src/Tools/ExternalAccess/OmniSharp.CSharp/Formatting/OmniSharpCSharpSyntaxFormattingOptions.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Formatting;
+using Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Formatting;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.CSharp.Formatting
+{
+    internal enum OmniSharpLabelPositionOptions
+    {
+        LeftMost = LabelPositionOptions.LeftMost,
+        OneLess = LabelPositionOptions.OneLess,
+        NoIndent = LabelPositionOptions.NoIndent
+    }
+
+    internal enum OmniSharpBinaryOperatorSpacingOptions
+    {
+        Single = BinaryOperatorSpacingOptions.Single,
+        Ignore = BinaryOperatorSpacingOptions.Ignore,
+        Remove = BinaryOperatorSpacingOptions.Remove
+    }
+
+    internal sealed record class OmniSharpCSharpSyntaxFormattingOptions(
```

Comment:
will we need to be proactive about adding to this if new features come in, or is it just on an as needed basis? If the former, can we have some way to verify or even generate this file?

Reply:
Since it is invoking the CSharpSyntaxFormattingOptions constructor we should see errors here if it is modified.

Reply:
Should be _as needed_.

Reply:
If the option is stored in editorconfig then it will be automatically picked up by `OmniSharpSyntaxFormattingOptionsWrapper.FromDocumentAsync` and will flow through. 
O# has an option to override/ignore editorconfig and in that case it would need to explicitly add a new option to the constructor if they want to pass it in.